src/engine/Test_engine.ml

```diff
@@ -327,10 +328,11 @@ let make_tests ?(unit_testing = false) ?(get_xlang = None)
       Parsing_stat.print_regression_information ~ext xs newscore;
     pr2 (spf "total mismatch: %d" !total_mismatch)
   in
-  (tests, print_summary)
+  (tests, !total_mismatch, print_summary)
```

Comment:
This will not work, you need to return the ref, otherwise you will always return 0
or maybe return a get_total_mismatch function that access the ref, to be more uniform with print_summary

Reply:
D'oh, thanks for catching this, I'll fix it in #9196.